File: myintent/src/main/java/pl/mareklangiewicz/myintent/MIExampleCmdListFragment.java
package pl.mareklangiewicz.myintent;

import android.os.Bundle;
import android.view.View;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Marek Langiewicz on 12.11.15.
 */
public class MIExampleCmdListFragment extends MICmdListFragment {
    @Override public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setImageRes(R.drawable.ic_find_replace_black_24dp);
        List<String> commands = new ArrayList<>();
        MIContract.CmdExample.load(getActivity(), commands);
        setCommands(commands);
    }
}

File: myintent/src/main/java/pl/mareklangiewicz/myintent/MIExampleCmdListFragment.kt
package pl.mareklangiewicz.myintent

import android.os.Bundle
import android.view.View

import java.util.ArrayList

/**
 * Created by Marek Langiewicz on 12.11.15.
 */
class MIExampleCmdListFragment : MICmdListFragment() {
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        manager?.name = "MI${if(BuildConfig.DEBUG) " D " else " "}${getString(R.string.mi_example_commands)}"
        imageRes = R.drawable.mi_ic_example_command_black_24dp
        refresh()
    }

    fun refresh() {
        var cmds = ArrayList<String>()
        MIContract.CmdExample.load(activity, cmds)
        commands = cmds
    }

}
